src/tests/electron/window-frame-listener.test.ts

```diff
@@ -20,75 +20,67 @@ describe(WindowFrameListener, () => {
 
     afterEach(() => {
         browserWindowMock.verifyAll();
+        windowStateActionsCreatorMock.verifyAll();
     });
 
     it('do nothing on action invocation before initialize', () => {
         browserWindowMock.setup(b => b.on(It.isAny(), It.isAny())).verifiable(Times.never());
     });
 
-    describe('initialize', () => {
-        it('registers to required window events', () => {
-            const eventCallbacks: Function[] = [];
+    describe('validate window listeners', () => {
+        let maximizeCallback: Function;
+        let unmaximizeCallback: Function;
+        let enterFullScreenCallback: Function;
+        let leaveFullScreenCallback: Function;
 
-            const eventsToListen = ['minimize', 'maximize', 'unmaximize'];
-            eventsToListen.forEach(eventName => {
-                browserWindowMock
-                    .setup(b => b.on(eventName as any, It.isAny()))
-                    .callback((event, cb) => {
-                        eventCallbacks.push(cb);
-                    })
-                    .verifiable(Times.once());
-            });
+        beforeEach(() => {
+            setupVerifiableWindowEventCallback('maximize', cb => (maximizeCallback = cb));
```

Comment:
should each test set mocks only for what each test need? 
Having all mocks set up for all cases make the code a little hard to follow as in, I have no idea which mock is really needed for each unit test.

Reply:
i made the browserWindowMock to be strict to control the access to browser window. So, I may need to do the mock on all the types anyway. Or i need to have 1 test with strict mock on initialize & then loose mock for others. Do you prefer that?